cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -730,64 +810,24 @@ Result<DeclarationInfo> FromProto(const substrait::Rel& rel, const ExtensionSet&
       std::vector<compute::Aggregate> aggregates;
       aggregates.reserve(measure_size);
       // store aggregate fields to be used when output schema is created
-      std::vector<std::vector<int>> agg_src_fieldsets(measure_size);
+      std::vector<std::vector<int>> agg_src_fieldsets;
```

Comment:
Why this change?

Reply:
For convenience. The next line reserves the same size. The only difference is that the vector is populated using `push_back` instead of `[i]`.